sahi/predict.py

```diff
@@ -538,11 +545,11 @@ def predict(
     for ind, image_path in enumerate(
         tqdm(image_iterator, f"Performing inference on {input_type_str}", total=num_frames)
     ):
-        # get filename
-        if source_is_video:
+        # Source is an image: Iterating over Image objects
+        if isinstance(image_path, Image.Image):
```

Comment:
this is incorrect, in this case source is a video
```suggestion
        if source and source_is_video:
```